src/main/java/org/opensearch/security/setting/SecurableLegacySetting.java

```diff
@@ -0,0 +1,99 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ *
+ * Modifications Copyright OpenSearch Contributors. See
+ * GitHub history for details.
+ */
+
+package org.opensearch.security.setting;
+
+import java.util.Optional;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+import org.opensearch.common.settings.SecureSetting;
+import org.opensearch.common.settings.Setting;
+import org.opensearch.common.settings.Settings;
+import org.opensearch.core.common.settings.SecureString;
+
+/**
+ * Wrapper for legacy settings that support a secure variant located in the Keystore.
+ * <p>
+ * Secure name is the insecure name with "_secure" appended to it.
+ */
+public class SecurableLegacySetting {
```

Comment:
I am a bit confused about the word "legacy" here. Could you please elaborate what "legacy settings" are?